Source/Connector.swift

```diff
@@ -97,7 +97,9 @@ class Connector {
                 let isConnected = strongSelf.connectedBox.read { $0.contains(peripheral.identifier) }
                 if isConnected {
                     strongSelf.disconnectingBox.write { $0.insert(peripheral.identifier) }
-                    strongSelf.centralManager.cancelPeripheralConnection(peripheral.peripheral)
+                    if strongSelf.centralManager.state == .poweredOn {
+                        strongSelf.centralManager.cancelPeripheralConnection(peripheral.peripheral)
```

Comment:
You should call also this line `strongSelf.disconnectingBox.write { $0.insert(peripheral.identifier) }` only when bluetooth is enabled.

Reply:
Done